projects/python-client/src/datapane/client/api/user.py

```diff
@@ -113,6 +116,76 @@ def _run_script(script: str):
     runpy.run_path(str(script_path), run_name="__datapane__")
 
 
+def _run_template(template_path: str):
+    """Run the template script"""
+    # Store the cwd before we change it
+    old_cwd = os.getcwd()
+    # Change cwd to support relative paths in template script
+    os.chdir(template_path)
+
+    try:
+        runpy.run_path(
+            "template.py",
+            run_name="__datapane__",
+        )
+    # Notify the user of missing packages that are required by the template
+    except ModuleNotFoundError as e:
+        raise DPError(f"Please install the following packages to run this template\n{e.name}") from e
+
+    os.chdir(old_cwd)
+
+
+def _download_template(url: str):
+    """Download the template from a repository url, and delete the .git directory
+
+    Returns:
+        The path of the cloned template repository
+    """
+
+    url = _check_repo_url(url)
+
+    # Check if target directory already exists
+    target = url.split("/")[-1]
+    target_dir_exists = os.path.exists(target)
+
+    # If directory exists, raise an error notifying the user
```

Comment:
The code is very readable here, so it already documents the "what".
Could the comments be refocused to talk about the "why" in your code?

i.e.
> avoid overwriting user's data

or
> present a clean error to the user, instead of letting them try to debug Git/dulwich themselves

Reply:
Updated to "Avoid overwriting user's data"